### PR TITLE
feat: diff-path-spec option

### PR DESCRIPTION
**Pull request by kgilpin**
By default, ignore _vendor_ and _node_modules_ when building patch file. This is customizable via diff-path-spec.

Without this option, the patch file that's created by the installer can generate huge patch files that include everything in  _node_modules_ and/or _vendor_ - if those directories aren't git-ignored.

### CODE REVIEW

File: test/smoke.spec.ts [hunk 1]
```diff
@@ -46,4 +49,13 @@ describe('install-appmap-action', () => {
...
     expect(patch.contents).toMatch(/\+install --no-interactive/);
   });
+
+  it('diff path spec can be configured', async () => {
+    installer.diffPathSpec = `. ':(exclude)install.log'`;
+    await installer.installAppMapTools();
+    await installer.installAppMapLibrary();
+    const patch = await installer.buildPatchFile();
+
+    expect(patch.contents).not.toMatch(/\+install --no-interactive/);
```
Comment (dividedmind) [inline test/smoke.spec.ts:59]: Should this also check if defaults aren't kept?

**Review by dividedmind — APPROVED** [review]

**Comment by dividedmind** [comment]
BTW, I wonder if it wouldn't be more user-friendly (albeit less flexible) to provide an `exclude` (optional) array option instead of the raw diff path spec.

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -48,27 +39,19 @@ jobs:
...
       - name: Archive AppMaps
-        if: ${{ github.event.pull_request.base.sha == '' }}
+        if: ${{ github.event_name != 'pull_request' }}
         uses: getappmap/archive-action@v1-pre.13
         with:
           verbose: true
 
       - name: Analyze AppMaps
-        if: ${{ github.event.pull_request.base.sha != '' }}
+        if: (success() || failure()) && ${{ github.event_name == 'pull_request' }}
```
[the remark this answers is not in the document]
Comment (kgilpin) [reply .github/workflows/main.yml:51]: But probably `always() && ${{ github.event_name == 'pull_request' }}` will be the same as `always()` which isn't what we want. We only want analysis to run on a PR-triggered build. LMK if I'm wrong about that...

**Comment by kgilpin** [comment]
> BTW, I wonder if it wouldn't be more user-friendly (albeit less flexible) to provide an `exclude` (optional) array option instead of the raw diff path spec.

Probably, but hopefully users almost never have to use this option. It's really just here to provide some way of fixing the patch file behavior if there's a problem.

File: .github/workflows/main.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (dividedmind) [reply .github/workflows/main.yml:51]: I don't think so, but [the docs](https://docs.github.com/en/actions/learn-github-actions/expressions#always) do say:
> If you want to run a job or step regardless of its success or failure, use the recommended alternative:if: success() || failure()

so maybe it is better this way.

**Pull request merged**